pandas/core/reshape/concat.py

```diff
@@ -619,10 +619,10 @@ def _make_concat_multiindex(indexes, keys, levels=None, names=None) -> MultiInde
         for hlevel, level in zip(zipped, levels):
             to_concat = []
             for key, index in zip(hlevel, indexes):
-                try:
-                    i = level.get_loc(key)
-                except KeyError as err:
-                    raise ValueError(f"Key {key} not in level {level}") from err
+                mask = level == key
+                if not any(mask):
```

Comment:
use mask.any()

Reply:
it’s a bit more idiomatic to use .get_indexer here which handles duplicates - if u can make that work

Reply:
@jreback thanks for the review. I've looked into that, but it seems it errors with duplicates:
```python
>>> pd.Index(['a', 'b']).get_indexer(['a'])                                 
array([0])
>>> pd.Index(['a', 'b', 'b']).get_indexer(['a'])
---------------------------------------------------------------------------
InvalidIndexError                         Traceback (most recent call last)
<ipython-input-10-0a1bc1d27a1b> in <module>
----> 1 pd.Index(['a', 'b', 'b']).get_indexer(['a'])

~/pandas-dev/pandas/core/indexes/base.py in get_indexer(self, target, method, limit, tolerance)
   2926 
   2927         if not self.is_unique:
-> 2928             raise InvalidIndexError(
   2929                 "Reindexing only valid with uniquely valued Index objects"
   2930             )

InvalidIndexError: Reindexing only valid with uniquely valued Index objects
```

Should I open an issue for `get_indexer` first and make sure that deals with duplicates before coming back to this one?